src/oc/web/actions.cljs

```diff
@@ -894,10 +894,37 @@
     (assoc-in db comments-key new-comments-data)))
 
 (defmethod dispatcher/action :comment-add/finish
-  [db [_ {:keys [activity-uuid]}]]
-  (api/get-comments activity-uuid)
+  [db [_ {:keys [activity-data]}]]
+  (api/get-comments activity-data)
   (assoc db :comment-add-finish true))
 
+(defmethod dispatcher/action :comment-delete
+  [db [_ activity-uuid comment-data]]
+  (api/delete-comment comment-data)
+  (let [org-slug (router/current-org-slug)
+        board-slug (router/current-board-slug)
+        item-uuid (:uuid comment-data)
+        comments-key (dispatcher/activity-comments-key org-slug board-slug activity-uuid)
+        comments-data (get-in db comments-key)
+        new-comments-data (remove #(= item-uuid (:uuid %)) comments-data)]
+    (assoc-in db comments-key new-comments-data)))
+
+(defmethod dispatcher/action :comment-delete/finish
+  [db [_]]
+  db)
```

Comment:
@thepug I see you send the `:success` value with `:comment-delete/finish` from oc.web.api. Maybe we can reload the comments if comment delete failed?
Since you have already removed the comment from the app-state object.
Every time a 5xx or a 0 is received from an api call a red banner is shown at the top of the page. With this we only need to make sure that the app-state is still in sync with the server.
If you don't like reloading comments you can wait to remove the comment from the app-state to here when you receive the response. Your call.